daffodil-core/src/main/scala/org/apache/daffodil/dsom/GlobalElementDecl.scala

```diff
@@ -28,18 +28,9 @@ final class GlobalElementDecl(
   with GlobalElementComponentMixin
   with ElementDeclFactoryDelegatingMixin
   with NestingTraversesToReferenceMixin
-  with ResolvesProperties // 
-    // Technically, this is not a term, so shouldn't resolve scoped properties
-  // and doesn't take into account combining ref with def properties.
-  // But we need to get one local property: dfdl:choiceBranchKey. In order to 
-  // enforce that it is NOT specified on global defs/decls.
-  // So we mixin ResolvesProperties, but shouldn't use it except to probe dfdl:choiceBranchKey
+  with ResolvesLocalProperties // TODO: what properties do we resolve locally on global elements?
```

Comment:
Did you figure out this TODO?

Reply:
No, I didn't look into it further. I think there are several that we look for a local definition of, if only to give warnings about. 

I'll search and update this.